lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -127,11 +127,32 @@ def get_current_span() -> Union[Span, None]:
     return cast(Span, span)
 
 
+def _get_tracer_from_context(ctx: Context) -> Optional[ContextVar]:
+    try:
+        return next(filter(lambda v: v is not None and v.name == "tracer", ctx.keys()))
+    except StopIteration:
+        return
+
+
 def _get_tracer() -> Optional[Tracer]:
+    """Find tracer in context variable and as a fallback locate it in the full context.
+
+    We need the fallback as `tracer` context variable isn't under the same reference in the instrumented function.
```

Comment:
not sure I understand what this means.

Reply:
Seems that comment wasn't relevant anymore. Removed the second line.